src/main/java/org/project_kessel/relations/client/HealthClient.java

```diff
@@ -0,0 +1,99 @@
+package org.project_kessel.relations.client;
+
+import java.util.logging.Logger;
+
+import io.grpc.Channel;
+import io.grpc.stub.StreamObserver;
+import io.grpc.stub.StreamObservers;
+import io.smallrye.mutiny.operators.multi.processors.UnicastProcessor;
+
+import org.project_kessel.api.relations.v1.GetLivezReply;
+import org.project_kessel.api.relations.v1.GetLivezRequest;
+import org.project_kessel.api.relations.v1.GetReadyzReply;
+import org.project_kessel.api.relations.v1.GetReadyzRequest;
+import org.project_kessel.api.relations.v1.KesselHealthGrpc;
+import io.smallrye.mutiny.Uni;
+
+public class HealthClient {
+    private static final Logger logger = Logger.getLogger(CheckClient.class.getName());
+
+    private final KesselHealthGrpc.KesselHealthStub asyncStub;
+    private final KesselHealthGrpc.KesselHealthBlockingStub blockingStub;
+
+
+    HealthClient(Channel channel) {
+        asyncStub = KesselHealthGrpc.newStub(channel);
+        blockingStub = KesselHealthGrpc.newBlockingStub(channel);
+    }
+
+    public void readyz(GetReadyzRequest request,
+     StreamObserver<GetReadyzReply> responseObserver) {
+        asyncStub.getReadyz(request, responseObserver);
+    }
+    
+     public void livez(GetLivezRequest request,
+     StreamObserver<GetLivezReply> responseObserver) {
+        asyncStub.getLivez(request, responseObserver);
+    }
+
+     public GetReadyzReply readyz(GetReadyzRequest request) {
+        return blockingStub.getReadyz(request);
+    }
+    
+     public GetLivezReply livez(GetLivezRequest request) {
+        return blockingStub.getLivez(request);
+    }
+
+    public Uni<GetReadyzReply> readyzUni(GetReadyzRequest request) {
```

Comment:
I think we can probably leave these `Uni` methods out. For liveness and readiness the simple blocking calls should be enough (though we might as well pass on the async StreamObserver calls as well). The reactive code seems like overkill here.

I do have it in for the CheckClient and check calls -- which in itself is a marginal call.

What do you think?

Reply:
I agree that its probably overkill for simple status requests, especially since there's only a single response and not needed to stream or wait for multiple objects to be returned.